### PR TITLE
Added make_props in AccordionTypes and onOpen prop

### PR DESCRIPTION
**Pull request by bhavyamm01**

### CODE REVIEW

File: src/components/generic/accordion/AccordionTypes.re [hunk 1]
```diff
@@ -6,4 +6,20 @@ type t = {
   header,
   footer: React.element,
   body: React.element,
+  onOpen: unit => unit,
+};
+
+let make_props =
+    (
+      ~header,
+      ~footer: React.element,
```
Comment (porter473) [inline src/components/generic/accordion/AccordionTypes.re:15]: use `React.null` for footer

File: examples/ExampleTable.re [hunk 2]
```diff
@@ -21,6 +21,21 @@ let make = () => {
...
+      header: Text("test"),
+      footer: <div />,
+      body: <div />,
+      onOpen: _ => Js.log("Accordion working")
+    },
+    {
+      header: Text("test2"),
+      footer: <div />,
+      body: <div />,
+      onOpen: _ => Js.log("Accordion working 2")
```
Comment (porter473) [inline examples/ExampleTable.re:35]: use `make_props`

File: examples/ExampleTable.re [hunk 3]
```diff
@@ -21,6 +21,21 @@ let make = () => {
...
+    }
```
Comment (porter473) [inline examples/ExampleTable.re:36]: use `make_props`

File: src/components/generic/accordion/Accordion.re [hunk 4]
```diff
@@ -17,32 +17,54 @@ let getAccordionBody = (body: React.element) => {
   <MaterialUi.AccordionDetails> body </MaterialUi.AccordionDetails>;
 };
 
+module AccordionModule = {
```
Comment (Prateek13727) [inline src/components/generic/accordion/Accordion.re:20]: rename to `AccordionWrapper`. The suffix Module is not adding any value to the name

File: src/components/generic/accordion/AccordionTypes.re [hunk 5]
```diff
@@ -6,4 +6,20 @@ type t = {
   header,
   footer: React.element,
   body: React.element,
+  onOpen: unit => unit,
+};
+
+let make_props =
+    (
+      ~header,
```
Comment (porter473) [inline src/components/generic/accordion/AccordionTypes.re:14]: Mention the `header` type. Lets make it a practice to define the function's parameter types

File: examples/ExampleTable.re (hunk 3 above)
Comment (porter473) [reply examples/ExampleTable.re:36]: If the added code is part of testing, then we can remove it.

**Review by porter473 — CHANGES_REQUESTED** [review]

**Pull request merged**